lib/livebook_web/live/session_live/cell_component.ex

```diff
@@ -270,6 +270,26 @@ defmodule LivebookWeb.SessionLive.CellComponent do
     live_component(socket, LivebookWeb.SessionLive.VegaLiteComponent, id: id, spec: spec)
   end
 
+  defp render_output(socket, {:kino_widget, :vega_lite, pid}, id) do
+    live_render(socket, LivebookWeb.Kino.VegaLiteLive,
+      id: id,
+      session: %{"id" => id, "pid" => pid}
+    )
+  end
+
+  defp render_output(_socket, {:kino_widget, type, _pid}, _id) do
+    formatted = """
+    Got unsupported Kino widget type: #{inspect(type)}, if that's a new widget
+    make usre to update Livebook to the latest version
+    """
+
+    assigns = %{formatted: formatted}
+
+    ~L"""
+    <div class="overflow-auto whitespace-pre text-red-600 tiny-scrollbar"><%= @formatted %></div>
+    """
+  end
```

Comment:
This looks great! We also need to fail for a format that we don't know.

Reply:
We know all possible formats, because the outputs are formatted by our [`DefaultFormatter`](https://github.com/elixir-nx/livebook/blob/main/lib/livebook/evaluator/default_formatter.ex). Or did I miss the point?

Reply:
I mean, we may add kino_v2 in the future and other messages. So we need to fail on unknown messages accordingly!

Reply:
Yeah, my point is that at any point all possible formats come from DefaultFormatter and the rendering is adjusted accordingly. For example in this PR we extend DefaultFormatter with `{:kino_widget, value.type, value.pid}` and we handle it here, if we were to remove that, we would do so in both places. Having a catch-all may be useful when developing new outputs, so that the page doesn't crash =)

Reply:
Ah, I see! As long as we don’t crash for unknown messages, we should be good!

Reply:
Oh, I see. The client will never send an unknown message in this case, because we are the ones loading the client.

Reply:
Exactly!